### PR TITLE
Removed no_sync argument from Bot#stop

### PR DESCRIPTION
**Pull request by fizvlad**
# Summary

Fix for #675 

Removed argument from `Bot#stop`. Changed documentation

### CODE REVIEW

**Comment by z64** [comment]
The argument needs to stay for a deprecation cycle, and the changelog needs to be updated to reflect that.

File: lib/discordrb/bot.rb [hunk 1]
```diff
@@ -257,9 +257,8 @@ def join
 
     # Stops the bot gracefully, disconnecting the websocket without immediately killing the thread. This means that
     # Discord is immediately aware of the closed connection and makes the bot appear offline instantly.
-    # @param no_sync [true, false] Whether or not to disable use of synchronize in the close method. This should be true if called from a trap context.
-    def stop(no_sync = false)
-      @gateway.stop(no_sync)
+    def stop
```
Comment (swarley) [inline lib/discordrb/bot.rb]:
```suggestion
    # @note This method no longer takes an argument as of 3.4.0 
    def stop(_no_sync = nil)
```

Comment (fizvlad) [reply lib/discordrb/bot.rb]: Shoud we use `@deprecated` insterad of `@note`?

Comment (swarley) [reply lib/discordrb/bot.rb]: unfortunately YARD does not give us a way to deprecate arguments, and would list the entire method as deprecated

Comment (z64) [reply lib/discordrb/bot.rb]: Yeah. It doesn't play well with linters and IDEs on the ecosystem - they yell at you that the whole method will be removed. I did it before and people weren't happy. :smile_cat:

**Comment by fizvlad** [comment]
> changelog needs to be updated to reflect that

I've removed `breaking change` prefix as this won't break 3.3.0 code.

**Pull request merged**